### PR TITLE
Clone a  note

### PR DESCRIPTION
**Pull request by Karuppayya**
Adding ability to clone a note.
New button on top of note to clone it
add_clone_button

Dialog asking for a name for the new note(clone)
clone_dialog

The cloned note
new_cloned_note

### CODE REVIEW

**Comment by r-kamath** [comment]
LGTM!
Thanks

**Comment by Leemoonsoo** [comment]
Tested and working nicely.

Current icon "icon-grid"
image
is not very intuitive to me.

What do you think "fa fa-copy"
image
?

Cloning notebook does not copy interpreter bindings. How about copying interpreter bindings, too?

**Comment by felixcheung** [comment]
+1 on copy icon and interpreter bindings

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 1]
```diff
@@ -35,8 +35,8 @@
 import org.apache.zeppelin.notebook.Notebook;
 import org.apache.zeppelin.notebook.Paragraph;
 import org.apache.zeppelin.scheduler.Job;
-import org.apache.zeppelin.scheduler.JobListener;
 import org.apache.zeppelin.scheduler.Job.Status;
+import org.apache.zeppelin.scheduler.JobListener;
```
Comment (mindprince) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: Revert this change.

**Comment by Karuppayya** [comment]
@Leemoonsoo The copy icon looks better :-).
@Leemoonsoo @felixcheung I have also added code to copy the interpreter bindings.

**Comment by jitenderaswani** [comment]
This is a much needed feature! Love it.

**Comment by Leemoonsoo** [comment]
LGTM!

**Comment by zenglinxi0615** [comment]
this feature sounds very good and helpful!  thanks for your work! Love it very much!
can I ask that, what's the schedule of this feature or is this work already finished?
when will this feature merge to master?

**Comment by Karuppayya** [comment]
@zenglinxi0615  the feature is complete. U can pull my branch to get hands on.
As to when it will be merged , tagging @Leemoonsoo

**Comment by Karuppayya** [comment]
Can we resolve the following issue.?
https://issues.apache.org/jira/browse/ZEPPELIN-126

**Comment by Leemoonsoo** [comment]
@Karuppayya could tell me your jira id?

**Comment by Karuppayya** [comment]
@Leemoonsoo  my id is karuppayya1990

**Comment by Leemoonsoo** [comment]
@Karuppayya can not find your jira apache id `karuppayya1990`. Could you confirm?

**Comment by Karuppayya** [comment]
Oops.. id is karup1990 @Leemoonsoo
